src/steps/git.rs

```diff
@@ -366,16 +386,25 @@ impl RepoStep {
         result
     }
 
-    /// Pull the repositories specified in `self.repos`.
+    /// Pulls or fetches the repositories specified in `self.repos`, depending on `fetch_only`.
     ///
     /// # NOTE
     /// This function will create an async runtime and do the real job so the
     /// function itself is not async.
-    fn pull_repos(&self, ctx: &ExecutionContext) -> Result<()> {
+    fn pull_or_fetch_repos(&self, ctx: &ExecutionContext) -> Result<()> {
+        let is_fetch_only = ctx.config().git_fetch_only();
+
         if ctx.run_type().dry() {
-            self.repos
-                .iter()
-                .for_each(|repo| println!("{}", t!("Would pull {repo}", repo = repo.display())));
+            self.repos.iter().for_each(|repo| {
+                println!(
+                    "{}",
+                    t!(
+                        "Would {action} {repo}",
+                        action = if is_fetch_only { "fetch" } else { "pull" },
+                        repo = repo.display()
```

Comment:
Have you verified that this works? The key is `Would fetch {repo}` and `Would pull {repo}`, not `Would {action} {repo}`. I would expect something like this (although this probably doesn't work):

```suggestion
                    t!(
                        format!("Would {} {{repo}}", if is_fetch_only { "fetch" } else { "pull" }),
                        repo = repo.display()
```